File: API/Models/Company.cs
using System.ComponentModel.DataAnnotations;

namespace Semsar_online.Models
{
    public class Company
    {
        [Key]
        public string Id { get; set; }
        public string Name {get; set;}
        public string? City { get; set; }
        public string? Address { get; set; }
        public string? Image { get; set; }
        public string? Description {get;set;}
        public ICollection<Property>? Properties { get; set; }
        public User? User { get; set; }
    }
}

File: frontend/src/app/Interfaces/company.ts
export interface companyProperty{
  id: number,
  numberOfRooms: number,
  area: number,
  price: number,
  Location: string,
  type: string,
  description: string,
  listingDate: string,
  downPayment: number,
  status: string
}
export interface Company {
  Id: string|undefined,
  City?: string|undefined,
  Address?: string|undefined,
  Image?: string|ArrayBuffer|null|undefined,
  Name?: string|undefined,
  Description?: string|undefined,
  Email?: string|undefined,
  WhatsApp?: string|undefined,
  Website?: string|undefined,
  Properties?: companyProperty[]
}
